packages/ra-ui-materialui/src/detail/TabbedShowLayout.js

```diff
@@ -86,13 +90,15 @@ export class TabbedShowLayout extends Component {
                 key={version}
                 {...sanitizeRestProps(rest)}
             >
-                <Tabs
-                    // The location pathname will contain the page path including the current tab path
-                    // so we can use it as a way to determine the current tab
-                    value={location.pathname}
-                    indicatorColor="primary"
-                >
-                    {Children.map(children, (tab, index) => {
+                {React.cloneElement(
+                    tabs,
+                    {
+                      // The location pathname will contain the page path including the current tab path
+                      // so we can use it as a way to determine the current tab
+                      value: location.pathname,
+                      match
+                    },
+                    Children.map(children, (tab, index) => {
```

Comment:
I'd put that into the TabbedShowLayoutTabs component. In my opinion, TabbedShowLayout should not mess with its grand-children (the children of its child Tabs).

Reply:
Should I pass `getTabFullPath` as a prop to TabbedShowLayoutTabs well, then? Or do I replicate it inside TabbedShowLayouTabs?

Reply:
As `getTabFullPath` will only be used in the child, it should be defined there.

Reply:
Done in https://github.com/marmelab/react-admin/pull/2740/commits/c2dd91d80f392fe0480745a0b2f9218d42f0fb26